src/lib/compress_hw.c

```diff
@@ -573,6 +574,31 @@ static int compress_hw_wait(void *data, int timeout_ms)
 	return oops(compress, EIO, "poll signalled unhandled event");
 }
 
+static int compress_hw_set_codec_params(void *data, struct snd_codec *codec)
+{
+	struct compress_hw_data *compress = (struct compress_hw_data *)data;
```

Comment:
why not pass argument as compress_hw_data, why void

Reply:
thanks for review,

This change is correct,  we should pass void *data as argument as it given in below ops structure.
reference link: https://github.com/alsa-project/tinycompress/blob/master/include/tinycompress/compress_ops.h#L18
struct compress_ops {
	int (*write)(void *compress_data, const void *buf, size_t size);
	int (*read)(void *compress_data, void *buf, size_t size);
	int (*start)(void *compress_data);
	int (*stop)(void *compress_data);
	int (*pause)(void *compress_data);
	int (*resume)(void *compress_data);
	int (*drain)(void *compress_data);
        int (*set_codec_params)(void *compress_data, struct snd_codec *codec);
}

thanks,
mohs